src/transport/AdminPairingTable.cpp

```diff
@@ -25,44 +25,100 @@
 #include <transport/AdminPairingTable.h>
 
 namespace chip {
+
+// TODO: Admin Pairing table should be backed by a single backing store (attribute store), remove delegate callbacks.
+namespace {
+PersistentStorageDelegate * gStorage             = nullptr;
```

Comment:
This seems to introduce significant coupling for the purpose of simplifying ember callbacks.

We should instead figure out how to either inject dependencies or accept that the storage method is a global and use it as such (e.g. KVS does this).

Reply:
Could we instead have the attribute store be saved/loaded from KVS? 

I am unsure how easy that is given that I expect some attributes to be operational rather than persisted. What is the plan to solve this?

Reply:
Here is what has been filed to follow up on this: https://github.com/project-chip/connectedhomeip/issues/6419

Once addOpCreds is implemented and attribute store persistence is implemented (is there an issue tracking this). The admin pairing table should be persisted only in attribute store. We need only one source of truth here so that the attribute store always reflects the fabrics list as is. I know the current way of doing it is not great but I did not want to spend too much time writing throwaway code considering all those delegate methods that update the attribute list will go away once we clean all this admin pairing code.

We need to:
1) Implement addOpCreds
2) Implement persistente for attribute store
3) Admin Pairing table should only modify fabrics attribute store: https://github.com/project-chip/connectedhomeip/issues/6419
4) Any read of the fabrics list would then be accurate.

Reply:
Resolved with Andrei